service/host.go

```diff
@@ -115,32 +116,29 @@ func (uis *UIServer) modifyHost(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// determine what action needs to be taken
-	switch opts.Action {
-	case "updateStatus":
-		currentStatus := h.Status
-		newStatus := opts.Status
-		if !util.StringSliceContains(validUpdateToStatuses, newStatus) {
-			http.Error(w, fmt.Sprintf("'%v' is not a valid status", newStatus), http.StatusBadRequest)
-			return
-		}
+	currentStatus := h.Status
+	modifyResult, restErr := modifyHostStatus(h, opts, u)
```

Comment:
nit, call `restErr` `err`

Reply:
Go complains since, by this point, it doesn't expect `err` to be specifically an `APIError` so it doesn't know that it has `StatusCode` and `Message` fields when they're referenced. Other places in the code make the type conversion more explicit, then refer to the converted error as an `apiErr`, would you recommend going down this route here too?

Reply:
Oh i didn't realise an `err error` was also in scope in here. Leave it be then